### PR TITLE
New data_source keycloak_realm

### PR DESCRIPTION
**Pull request by marcoreni**
This adds a new datasource for keycloak_realm.

Acceptance test results:
```
=== RUN   TestAccKeycloakDataSourceRealm_basic
--- PASS: TestAccKeycloakDataSourceRealm_basic (0.72s)
```

### CODE REVIEW

**Comment by marcoreni** [comment]
CircleCI failure due to https://github.com/mrparkers/terraform-provider-keycloak/issues/32

File: docs/data_sources/keycloak_realm.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+### Argument Reference
+
+The following arguments are supported:
+
+- `realm` - (Required) The realm name.
+
+### Attributes Reference
+
+See the [Realm](../resources/keycloak_realm.md) for details on the returned attributes.
```
Comment (mrparkers) [inline docs/data_sources/keycloak_realm.md]:
```suggestion
See the docs for the [`keycloak_realm` resource](../resources/keycloak_realm.md) for details on the exported attributes.
```

File: mkdocs.yml [hunk 2]
```diff
@@ -4,6 +4,9 @@ repo_url: https://github.com/mrparkers/terraform-provider-keycloak
 nav:
 - Getting Started: index.md
 - Data Sources:
+  - keycloak_openid_client: data_sources/keycloak_openid_client.md
+  - keycloak_openid_client_authorization_policy: data_sources/keycloak_openid_client_authorization_policy.md
```
Comment (mrparkers) [inline mkdocs.yml]: So this is awkward, but there actually aren't doc files for these data sources you added.  I'd appreciate a PR to add these if you're interested :smile:

If not, could you remove these dead links?

Comment (marcoreni) [reply mkdocs.yml]: Whops! My mistake, I planned on adding those documents but I read that you were working on it so I deleted the files, but I forgot to delete the links :D

I'll file a separate PR soon, for now I'm deleting the references.

**Comment by mrparkers** [comment]
LGTM, thanks for the PR!  Sorry again about the flaky tests

**Pull request merged**